setup.py

```diff
@@ -1,135 +1,23 @@
-#!/usr/bin/env python
+import subprocess
 
-from wagtail import __version__
-from wagtail.utils.setup import assets, check_bdist_egg, sdist
+from setuptools import setup
+from setuptools.command.sdist import sdist as base_sdist
 
-try:
-    from setuptools import find_packages, setup
-except ImportError:
-    from distutils.core import setup
 
+class sdist(base_sdist):
+    def compile_assets(self):
+        try:
+            subprocess.check_call(["npm", "run", "build"])
+        except (OSError, subprocess.CalledProcessError) as e:
+            print("Error compiling assets: " + str(e))  # noqa: T201
+            raise SystemExit(1)
 
-# Hack to prevent "TypeError: 'NoneType' object is not callable" error
-# in multiprocessing/util.py _exit_function when setup.py exits
-# (see http://www.eby-sarna.com/pipermail/peak/2010-May/003357.html)
-try:
-    import multiprocessing  # noqa: F401
-except ImportError:
-    pass
+    def run(self):
+        self.compile_assets()
+        base_sdist.run(self)
```

Comment:
Looking at this again, since Python 3 this should simply be

```suggestion
        super().run()
````

I've made the change locally and did e2e testing again, and it still works well. Will push the change as I merge